src/AWSCredentials.jl

```diff
@@ -185,263 +190,311 @@ function localhost_is_ec2()
-    creds = nothing
+function dot_aws_config(profile=nothing)
     config_file = dot_aws_config_file()
 
-    ini = nothing
     if isfile(config_file)
         ini = read(Inifile(), config_file)
-        p = profile === nothing ? aws_get_profile() : profile
-        key, key_id, token = aws_get_credential_details(p, ini, true)
+        p = profile === nothing ? _aws_get_profile() : profile
+        println(typeof(profile))
+        access_key, secret_key, token = _aws_get_credential_details(p, ini, true)
 
-        if key !== :notfound
-            creds = AWSCredentials(key_id, key, token)
+        if access_key !== :notfound
+            return AWSCredentials(access_key, secret_key, token)
         else
-            creds = aws_get_role(p, ini)
+            return _aws_get_role(p, ini)
         end
     end
 
-    return creds
+    return nothing
 end
 
-function aws_get_role_details(profile::AbstractString, ini::Inifile)
-    if debug_level > 0
-        println("Loading \"$profile\" Profile from " *
-                dot_aws_config_file() * "... ")
-    end
-
-    role_arn = get(ini, profile, "role_arn")
-    source_profile = get(ini, profile, "source_profile")
-
-    profile = "profile $profile"
-    role_arn = get(ini, profile, "role_arn", role_arn)
-    source_profile = get(ini, profile, "source_profile", source_profile)
 
-    (source_profile, role_arn)
-end
+"""
+    _aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
 
-function aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
-    if debug_level > 0
-        filename = config ? dot_aws_config_file() : dot_aws_credentials_file()
-        println("Loading \"$profile\" AWSCredentials from " * filename
-                * "... ")
-    end
+# Arguments
+- `profile::AbstractString`: Specific profile used to get AWSCredentials
+- `ini::Inifile`: Inifile to look into for the `profile` credentials
+- `config::Bool`: True if using `~/.aws/config`
 
-    key_id = get(ini, profile, "aws_access_key_id")
-    key = get(ini, profile, "aws_secret_access_key")
+Get `AWSCredentials` for the specified `profile` from the `inifile`. If targeting the `~/.aws/config`
+file, with a non-default `profile`, you must specify `config=true` otherwise the default credentials
+will be returned.
+"""
+function _aws_get_credential_details(profile::AbstractString, ini::Inifile, config::Bool)
+    access_key = get(ini, profile, "aws_access_key_id")
+    secret_key = get(ini, profile, "aws_secret_access_key")
     token = get(ini, profile, "aws_session_token", "")
 
     if config
         profile = "profile $profile"
-        key_id = get(ini, profile, "aws_access_key_id", key_id)
-        key = get(ini, profile, "aws_secret_access_key", key)
+        access_key = get(ini, profile, "aws_access_key_id", access_key)
+        secret_key = get(ini, profile, "aws_secret_access_key", secret_key)
         token = get(ini, profile, "aws_session_token", token)
     end
```

Comment:
An `aws_session_token` is only required if you specify temporary credentials, and is not used otherwise (long-term credentials).

This logic definitely is not correct, however I'm currently inclined to log an issue for this as I want to re-write most of this code and it is not a breaking issue.

What are your thoughts?

---

For posterity, and anyone in the future looking at this, the issue broken down:
```
[test]
aws_access_key_id=TEMP_ACCESS_KEY
aws_secret_access_key=TEMP_SECRET_KEY
aws_session_token=TEMP_SESSION_TOKEN

[profile test]
aws_access_key_id=LT_ACCESS_KEY
aws_secret_access_key=LT_SECRET_KEY
```

Returns:
```
access_key=LT_ACCESS_KEY
secret_key=LT_SECRET_KEY
token=TEMP_SESSION_TOKEN  # Should be ""
```